### PR TITLE
feat(dev): enable the usage of the command property without port

### LINKED ISSUE #3654: Support arbitrary dev commands alongside static server
**Is your feature request related to a problem? Please describe.**

Some frameworks don't use a dev server, but do have a command to watch source files and compile them to static files and/or functions. These watch commands can't be specified as the `dev` command because they don't open a port, and because specifying a command disables the static server.

**Describe the solution you'd like**

A way to specify a non-server `dev` command in toml and framework-info that would run as a long-running process alongside the static and functions servers. This could be a simple as making this the behaviour if no target port has been specified.

**Describe alternatives you've considered**

Telling users to run the `watch` command in a separate terminal to the `ntl dev` command. Using `npm-run-all` or equivalent.

**Additional context**
This is inspired by Remix, which needs this.

**Can you submit a pull request?**

Yes

> Pull requests are welcome! If you would like to help us add this feature, please check our
> [contributions guidelines](../../CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by lukasholzer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #3654

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/utils/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+export type FrameworkNames = '#static' | '#auto' | '#custom'
```
Comment (ascorbic) [inline src/utils/types.d.ts:1]: Framework names can also be arbitrary strings

**Review by ascorbic — COMMENTED** [review]
This is looking good so far, and works for me locally with Remix. The behaviour is a significant change though, so you should ensure that you test various scenarios to avoid breaking existing sites.

File: src/utils/types.d.ts (hunk 1 above)
Comment (lukasholzer) [reply src/utils/types.d.ts]: thx updated :)

**Comment by ascorbic** [comment]
This will probably need a docs update, explaing that if targetPort is provided then it will wait for that port to open, and proxy to it, and if it's not provided then the static server will be used.

**Comment by lukasholzer** [comment]
> This will probably need a docs update, explaing that if targetPort is provided then it will wait for that port to open, and proxy to it, and if it's not provided then the static server will be used.

I think it should be updated here as well: https://docs.netlify.com/configure-builds/file-based-configuration/#netlify-dev

Hey as the docs in the repo are "autogenerated" `<!-- AUTO-GENERATED-CONTENT:START (GENERATE_COMMANDS_DOCS) -->` – can you point me in the direction on where to adapt it?

**Comment by ascorbic** [comment]
What's needed to land this? It would be great to get this out, now that Remix has been released

**Comment by lukasholzer** [comment]
@ascorbic I'm opening up a PR on the docs site as well to change it but IMHO we can release it first ;) From my point, it should be good to go but I'm the PR issuer so maybe @ehmicky can provide here more information.

**Comment by ascorbic** [comment]
Great! Could you complete the PR description now it's ready for review?

**Review by ehmicky — APPROVED** [review]

**Pull request merged**